src/ucp/wireup/stub_ep.c

```diff
@@ -416,3 +429,19 @@ int ucp_stub_ep_test(uct_ep_h uct_ep)
 {
     return uct_ep->iface == &ucp_stub_iface;
 }
+
+int ucp_stub_ep_test_aux(uct_ep_h stub_ep, uct_ep_h aux_ep)
+{
+    return ucp_stub_ep_test(stub_ep) &&
+           (ucs_derived_of(stub_ep, ucp_stub_ep_t)->aux_ep == aux_ep);
+}
+
+void ucp_stub_ep_extract_aux(uct_ep_h *uct_ep)
+{
+    uct_ep_h ep            = *uct_ep;
+    ucp_stub_ep_t *stub_ep = ucs_derived_of(ep, ucp_stub_ep_t);
+
+    *uct_ep = stub_ep->aux_ep;
+    stub_ep->aux_ep = NULL;
+    uct_ep_destroy(ep);
```

Comment:
this should be done by the caller - stub_ep should just extract aux_ep, not destroy user's ep